lib/shared.bash

```diff
@@ -220,6 +230,21 @@ function build_image_override_file_with_version() {
       done
     fi
 
+    if [[ "$cache_to_amt" -gt 0 ]] ; then
+      if ! docker_compose_supports_cache "$version" ; then
+        echo "Unsupported Docker Compose config file version: $version"
+        echo "The 'cache_to' option can only be used with Compose file versions 2.2 or 3.2 and above."
+        echo "For more information on Docker Compose configuration file versions, see:"
+        echo "https://docs.docker.com/compose/compose-file/compose-versioning/#versioning"
+        exit 1
+      fi
+
+      printf "      cache_to:\\n"
+      for cache_to_i in "${cache_to[@]}"; do
+        printf "        - %s\\n" "${cache_to_i}"
+      done
+    fi
+
```

Comment:
optional: may make sense to refactor this to its own function and use it twice instead of copy-pasting :P

Reply:
- [x] refactored